tests/src/JIT/Regression/JitBlue/GitHub_17585/GitHub_17585.il

```diff
@@ -0,0 +1,184 @@
+// unmanaged, in this case it is JIT_TailCallHelperStub_ReturnAddress from tail call optimization.
+// To repro this issue on amd64/arm64 it also requires to reject fast tail call, for this puprose
+// `IMyComparer::Compare` has several stack arguments.
+
+// The fix is to always generate explicit null check before calling VSD with tail call from tail call frame.
+
+
+.assembly extern System.Runtime
+{
+}
+.assembly extern System.Runtime.Extensions
+{
+}
+.assembly GitHub_17585
+{
+}
+
+
+// =============== CLASS MEMBERS DECLARATION ===================
+
+.class interface private abstract auto ansi IMyComparer
+{
+  .method public hidebysig newslot abstract virtual
+          instance int32  Compare(object x,
+                                  object y,
+                                  int32 a,
+                                  int32 b,
+                                  int32 c,
+                                  int32 d) cil managed
+  {
+  } // end of method IMyComparer::Compare
+
+} // end of class IMyComparer
+
+.class private auto ansi beforefieldinit MyComparer`1<T>
+       extends [System.Runtime]System.Object
+       implements IMyComparer
+{
+  .method public hidebysig newslot virtual final
+          instance int32  Compare(object x,
+                                  object y,
+                                  int32 a,
+                                  int32 b,
+                                  int32 c,
+                                  int32 d) cil managed
+  {
+    // Code size       2 (0x2)
+    .maxstack  8
+    IL_0000:  ldc.i4.1
+    IL_0001:  ret
+  } // end of method MyComparer`1::Compare
+
+  .method public hidebysig specialname rtspecialname
+          instance void  .ctor() cil managed
+  {
+    // Code size       7 (0x7)
+    .maxstack  8
+    IL_0000:  ldarg.0
+    IL_0001:  call       instance void [System.Runtime]System.Object::.ctor()
+    IL_0006:  ret
+  } // end of method MyComparer`1::.ctor
+
+} // end of class MyComparer`1
+
+.class private auto ansi beforefieldinit MyTuple
+       extends [System.Runtime]System.Object
+{
+  .field private int16 m_value
+  .method public hidebysig instance int32
+          CompareTo(object other,
+                    class IMyComparer comparer) cil managed
+  {
+    // Code size       40 (0x28)
+    .maxstack  7
+    .locals init ([0] class MyTuple objTuple)
+    IL_0000:  ldarg.1
+    IL_0001:  isinst     MyTuple
+    IL_0006:  stloc.0
+    IL_0007:  ldarg.2
+    IL_0008:  ldarg.0
+    IL_0009:  ldfld      int16 MyTuple::m_value
+    IL_000e:  box        [System.Runtime]System.Int16
+    IL_0013:  ldloc.0
+    IL_0014:  ldfld      int16 MyTuple::m_value
+    IL_0019:  box        [System.Runtime]System.Int16
+    IL_001e:  ldc.i4.1
+    IL_001f:  ldc.i4.2
+    IL_0020:  ldc.i4.3
+    IL_0021:  ldc.i4.4
+    IL_0022:  tail. callvirt   instance int32 IMyComparer::Compare(object,
```

Comment:
These method names are misleading. It would be better to use method names that capture aspects important for the tests. 

For example, this method can be called `MethodWithStackArguments`. And you make it take even more arguments than this so that it tests the desired codepath even on ARM64.

Reply:
Fixed